app/client/components/delivery/address/deliveryAddressEditConfirmation.tsx

```diff
@@ -1,115 +1,318 @@
+          </PageNavAndContentContainer>
+        </>
       ) : (
         visuallyNavigateToParent(props)
       )}
     </WizardStep>
   );
 };
 
-interface AddressDisplayContainerProps {
-  title: string;
-  children: ReactElement<any> | null;
-}
-const AddressDisplayContainer = (props: AddressDisplayContainerProps) => (
-  <section
-    css={css`
-      border: 1px solid ${palette.neutral["86"]};
-      padding: 48px 0 14px;
-      position: relative;
-      section {
-        margin: 14px 0;
-        padding: 0 14px;
-        position: relative;
-      }
-      section + section {
-        padding-top: 14px;
-        border-top: 1px solid ${palette.neutral["86"]};
-      }
-    `}
-  >
-    <h1
-      css={css`
-        width: 100%;
-        position: absolute;
-        top: 0;
-        left: 0;
-        margin: 0;
-        padding: 0 14px;
-        ${textSans.medium()};
-        font-weight: bold;
-        line-height: 48px;
-        background-color: ${palette.neutral["97"]};
-        border-bottom: 1px solid ${palette.neutral["86"]};
-      `}
-    >
-      New address details
-    </h1>
-    <>{props.children}</>
-  </section>
-);
+export const DeliveryAddressEditConfirmation = (props: RouteableStepProps) => {
+  const addressContext = useContext(NewDeliveryAddressContext);
+  const contactIdContext = useContext(ContactIdContext);
+  const addressChangedInformationContext = useContext(
+    AddressChangedInformationContext
+  );
+
+  const addressChangeEmailCopy = [
+    ...addressChangedInformationContext
+      // tslint:disable-next-line: no-bitwise
+      .filter((_, i) => !(i & 1))
+      .map((element, index) => {
+        const originalAffectedDate =
+          addressChangedInformationContext[index * 2 + 1].value;
+        return {
+          name: element.title,
+          subId: element.value,
+          date:
+            originalAffectedDate && originalAffectedDate !== "-"
+              ? `as of front cover dated ${moment(
+                  `${originalAffectedDate}`,
+                  "D MMM YYYY"
+                ).format("dddd Do MMMM YYYY")}`
+              : ""
+        };
+      })
+      .map(
+        element =>
+          `${element.name} subscription (${element.subId}) ${element.date}`
+      ),
+    "",
+    `(as displayed on confirmation page at ${moment().format(
+      "H:mm:ss zz [on] Do MMMM YYYY"
+    )})`
+  ].join("\n");
+
+  return addressContext.newDeliveryAddress ? (
+    <AsyncLoader
+      render={renderConfirmation(props)}
+      fetch={updateAddressFetcher(
+        {
+          ...addressContext.newDeliveryAddress,
+          addressChangeInformation: addressChangeEmailCopy
+        },
+        contactIdContext
+      )}
+      readerOnOK={(resp: Response) => resp.text()}
+      loadingMessage={"Updating delivery address details..."}
+    />
+  ) : (
+    <GenericErrorScreen loggingMessage="delivery address confirmation page - no new delivery address context" />
```

Comment:
is this not a scenario where we just want to do the `visuallyNavigateToParent` thing??